pylearn2/models/multisubject.py

```diff
@@ -555,21 +575,25 @@ def cost_from_X(self, data):
         ----------
         data : WRITEME
         """
-        self.cost_from_X_data_specs()[0].validate(data)
-        X, Y = data
-        Y_hat = self.fprop(X)
-        return self.cost(Y, Y_hat)
+
+        for mlp, ds in safe_izip(model.mlps, data): 
+            self.cost_from_X_data_specs()[0].validate(data)
+            X, Y = data
+            Y_hat = self.fprop(X)
+            return self.cost(Y, Y_hat)
```

Comment:
This method should accumulate the costs from each mlp. I would write it in a similar way to the Default with something like:
```
cost = None
for loop over mlps:
    c = mlp.cost()
    if cost is None:
        cost = c
    else:
        cost = cost + weight * c
```